compiler/plc_project/src/build_config.rs

```diff
@@ -34,25 +36,38 @@ pub enum LinkageInfo {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
 pub struct ProjectConfig {
     pub name: String,
     pub files: Vec<PathBuf>,
     #[serde(default)]
     pub compile_type: FormatOption,
     #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub output: Option<String>,
     #[serde(default)]
     pub libraries: Vec<LibraryConfig>,
     #[serde(default)]
     pub package_commands: Vec<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub version: Option<String>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(alias = "format-version")]
+    #[serde(rename(serialize = "format-version"))]
+    pub format_version: Option<String>,
 }
 
 impl ProjectConfig {
     /// Retuns a project from the given string (in json format)
     /// All environment variables (marked with `$VAR_NAME`) that can be resovled at this time are resolved before the conversion
     pub fn try_parse(content: &str) -> Result<Self, Diagnostic> {
         let content = resolve_environment_variables(content)?;
-        serde_json::from_str(&content).map_err(Into::into)
+        let content: ProjectConfig = serde_json::from_str(&content).map_err(Diagnostic::from)?;
+        content.validate()?;
```

Comment:
It would be nice if we could also provide the schema as optional param. I don't see a reason to disable validation here, but maybe just let the user control it a bit more with a custom schema.

Not necessary for now, just idle thoughts